csrc/python_frontend/fusion_definition.cpp

```diff
@@ -130,14 +131,38 @@ void FusionDefinition::setupSchedule(const at::ArrayRef<c10::IValue>& inputs) {
 void FusionDefinition::finalizeSchedule(
     const at::ArrayRef<c10::IValue>& inputs) {
   FUSER_PERF_SCOPE("FusionDefinition::finalizeSchedule");
+#ifdef NVFUSER_DISTRIBUTED
+  // TODO: remove when multidevice executor integration is done natively
+  Fusion* fusion = user_sched_->schedule.get();
+  std::vector<Val*> fusion_inputs = InputsOf::outputs(fusion->outputs());
+  std::vector<Val*> vals = DependencyCheck::getAllValsBetween(
```

Comment:
Would this work? https://github.com/NVIDIA/Fuser/blob/618c4b79a053692bf83ebb32aa19dcc907ebb534/csrc/ir/utils.cpp#L406